frame/offences/src/tests.rs

```diff
@@ -71,14 +75,34 @@ fn should_not_report_the_same_authority_twice_in_the_same_slot() {
 	});
 }
 
+#[test]
+fn should_not_report_an_obsolete_offence() {
+	new_test_ext().execute_with(|| {
+		// given
+		let time_slot = 42;
+		let session_index = 3;
+		assert_eq!(offence_reports(KIND, time_slot), vec![]);
+
+		let offence =
+			Offence { validator_set_count: 5, session_index, time_slot, offenders: vec![5] };
+		assert_eq!(Offences::report_offence(vec![], offence), Err(OffenceError::ObsoleteReport));
+
+		with_on_offence_fractions(|f| {
+			assert_eq!(f.clone(), vec![]);
+		});
+	});
+}
+
 #[test]
 fn should_report_in_different_time_slot() {
 	new_test_ext().execute_with(|| {
 		// given
 		let time_slot = 42;
+		let session_index = 10;
```

Comment:
You could put `pub static SessionIndex: u32 = 10` and just use that across the board, so that in case you change it it would be fine. 

Also, you could then consider writing some tests in which the current session changes.

Reply:
The test `should_properly_clear_obsolete_offences` actually changes the session_index. It calls `on_session_change` with session_index as `SESSION_INDEX + 5`.